Source/WebKit/WebProcess/cocoa/WebProcessCocoa.mm

```diff
@@ -887,7 +887,8 @@ static void registerLogHook()
     WebCore::PublicSuffixStore::singleton().enablePublicSuffixCache();
 
 #if ENABLE(LOGD_BLOCKING_IN_WEBCONTENT)
-    if (isLockdownModeEnabled()) {
+    // We cannot use isLockdownModeEnabled() here, since it has not been set to the correct value yet.
+    if (WTF::processHasEntitlement("com.apple.private.webkit.lockdown-mode"_s)) {
```

Comment:
Let's just initialize `m_isLockdownModeEnabled` earlier in `WebProcess::initializeWebProcess()` instead to avoid bugs like this.

Reply:
Earlier as in:
before the call to `platformInitializeWebProcess(parameters);`